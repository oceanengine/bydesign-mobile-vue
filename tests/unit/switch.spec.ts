import { mount } from '@vue/test-utils';
import { Switch as NativeSwitch } from '@src'
const Switch = NativeSwitch as any;

describe('Switch', () => {
  it('renders switch when passed', () => {
    const wrapper = mount(Switch as any, {
      propsData: {
        value: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('renders switch when disabled', () => {
    const wrapper = mount(Switch as any, {
      propsData: {
        disabled: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('click event', () => {
    const onChange = jest.fn();
    const wrapper = mount(Switch, {
      context: {
        on: {
          change: onChange,
        },
      },
    });
    const dom = wrapper.find('.byted-switch');
    dom.trigger('click');
    expect(onChange).toHaveBeenCalled();
  });

  test('Control events', () => {
    const onChange = jest.fn();
    const wrapper = mount({
      template: `
        <byted-switch :value="value"  @input="onChange" />
      `,
      data() {
        return {
          value: false,
        };
      },
      methods: {
        onChange,
      },
    }) as any;
    const dom = wrapper.find('.byted-switch');
    dom.trigger('click');
    expect(onChange).toHaveBeenCalled();
    expect(wrapper.vm.value).toBe(false);
  });
});
