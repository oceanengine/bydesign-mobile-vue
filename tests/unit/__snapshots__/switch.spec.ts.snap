// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders switch when disabled 1`] = `
<div class="byted-switch byted-switch--disabled normal">
  <div class="byted-switch__show"></div>
</div>
`;

exports[`Switch renders switch when passed 1`] = `
<div class="byted-switch byted-switch--checked normal">
  <div class="byted-switch__show"></div>
</div>
`;
